### ISSUE TITLE
nvidia-smi output does not indicate gpu is being utilized

### ISSUE THREAD

**Issue by ipvirg** [mlperf/training_results_v0.6#18]
Launching the training jobs completes fairly quick 
sudo DATADIR=/home/virgil/training_results_v0.6/NVIDIA/benchmarks/gnmt/implementations/data LOGDIR=/home/virgil/training_results_v0.6/NVIDIA/benchmarks/gnmt/implementations/logs DGXSYSTEM=SUT PULL=0 ./run.sub

With a run time of 3-4 seconds each run for 10 runs.

200609191540237787829_10.log:RESULT,RNN_TRANSLATOR,,3,nvidia,2020-06-09 07:19:23 PM
200609191540237787829_1.log:RESULT,RNN_TRANSLATOR,,4,nvidia,2020-06-09 07:17:57 PM
200609191540237787829_2.log:RESULT,RNN_TRANSLATOR,,3,nvidia,2020-06-09 07:18:07 PM
200609191540237787829_3.log:RESULT,RNN_TRANSLATOR,,4,nvidia,2020-06-09 07:18:16 PM
200609191540237787829_4.log:RESULT,RNN_TRANSLATOR,,3,nvidia,2020-06-09 07:18:26 PM
200609191540237787829_5.log:RESULT,RNN_TRANSLATOR,,4,nvidia,2020-06-09 07:18:35 PM
200609191540237787829_6.log:RESULT,RNN_TRANSLATOR,,3,nvidia,2020-06-09 07:18:45 PM
200609191540237787829_7.log:RESULT,RNN_TRANSLATOR,,4,nvidia,2020-06-09 07:18:54 PM
200609191540237787829_8.log:RESULT,RNN_TRANSLATOR,,3,nvidia,2020-06-09 07:19:04 PM
200609191540237787829_9.log:RESULT,RNN_TRANSLATOR,,4,nvidia,2020-06-09 07:19:13 PM

While the training jobs were running, nvidia-smi command is not indicating gpu is being utilized
~/training_results_v0.6/NVIDIA/benchmarks/gnmt/implementations/logs$ nvidia-smi
Tue Jun  9 19:38:30 2020
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.64.00    Driver Version: 440.64.00    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla M60           Off  | 00000001:00:00.0 Off |                  Off |
| N/A   39C    P0    37W / 150W |      0MiB /  8129MiB |      5%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

Here's the a sample output on one of the ten logs -
Beginning trial 1 of 10
Gathering sys log on virg-rg-mlperf
:::MLL 1591730273.089 submission_benchmark: {"value": "gnmt", "metadata": {"file": "mlperf_log_utils.py", "lineno": 176}}
:::MLL 1591730273.090 submission_org: {"value": "NVIDIA", "metadata": {"file": "mlperf_log_utils.py", "lineno": 181}}
WARNING: Log validation: Key "submission_division" is not in known gnmt keys.
:::MLL 1591730273.090 submission_division: {"value": "closed", "metadata": {"file": "mlperf_log_utils.py", "lineno": 185}}
:::MLL 1591730273.091 submission_status: {"value": "onprem", "metadata": {"file": "mlperf_log_utils.py", "lineno": 189}}
:::MLL 1591730273.091 submission_platform: {"value": "1xVirtual Machine", "metadata": {"file": "mlperf_log_utils.py", "lineno": 193}}
:::MLL 1591730273.092 submission_entry: {"value": "{'hardware': 'Virtual Machine', 'framework': 'PyTorch NVIDIA Release 19.05', 'power': 'N/A', 'notes': 'N/A', 'interconnect': ' ', 'os': 'Ubuntu 16.04.6 LTS / ', 'libraries': \"{'container_base': 'Ubuntu-16.04', 'openmpi_version': '3.1.3', 'mofed_version': '4.4-1.0.1', 'cuda_version': '10.1.163', 'cuda_driver_version': '418.67', 'nccl_version': '2.4.6', 'cudnn_version': '7.6.0.64', 'cublas_version': '10.2.0.163', 'trt_version': '5.1.5.0', 'dali_version': '0.9.1'}\", 'compilers': 'gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609', 'nodes': \"{'num_nodes': '1', 'cpu': '1x Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz', 'num_cores': '6', 'num_vcpus': '6', 'accelerator': 'Tesla M60', 'num_accelerators': '1', 'sys_mem_size': '55 GB', 'sys_storage_type': 'SATA SSD', 'sys_storage_size': '1x 340G + 1x 100G', 'cpu_accel_interconnect': 'QPI', 'network_card': '', 'num_network_cards': '0', 'notes': ''}\"}", "metadata": {"file": "mlperf_log_utils.py", "lineno": 197}}
:::MLL 1591730273.092 submission_poc_name: {"value": "<removed>", "metadata": {"file": "mlperf_log_utils.py", "lineno": 201}}
:::MLL 1591730273.093 submission_poc_email: {"value": "<removed>", "metadata": {"file": "mlperf_log_utils.py", "lineno": 205}}
Clearing caches
:::MLL 1591730276.911 cache_clear: {"value": true, "metadata": {"file": "<string>", "lineno": 1}}
Launching on node virg-rg-mlperf
+ pids+=($!)
+ set +x
++ eval echo
+++ echo
+ docker exec -e DGXSYSTEM=SUT -e 'MULTI_NODE= --master_port=5024' -e LR=2.0e-3 -e TRAIN_BATCH_SIZE=256 -e TEST_BATCH_SIZE=128 -e WARMUP_STEPS=200 -e REMAIN_STEPS=6453 -e DECAY_INTERVAL=809 -e TARGET=24.0 -e NUMEPOCHS=8 -e MAX_SEQ_LEN=75 -e 'EXTRA_OPTS=   --fused-attention    --fused-xentropy    --no-log-all-ranks    ' -e SLURM_JOB_ID=200609191540237787829 -e SLURM_NTASKS_PER_NODE= -e SLURM_NNODES=1 cont_200609191540237787829 ./run_and_time.sh
Run vars: id 200609191540237787829 gpus 1 mparams  --master_port=5024
STARTING TIMING RUN AT 2020-06-09 07:17:57 PM
+ DATASET_DIR=/data
+ PREPROC_DATADIR=/preproc_data
+ RESULTS_DIR=gnmt_wmt16
+ LR=2.0e-3
+ TRAIN_BATCH_SIZE=256
+ TEST_BATCH_SIZE=128
+ WARMUP_STEPS=200
+ REMAIN_STEPS=6453
+ DECAY_INTERVAL=809
+ TARGET=24.0
+ MAX_SEQ_LEN=75
+ NUMEPOCHS=8
+ EXTRA_OPTS='   --fused-attention    --fused-xentropy    --no-log-all-ranks    '
+ BIND_LAUNCH=1
+ MATH=amp_fp16
+ [[ 1 -eq 1 ]]
+ LAUNCH_OPT='bind_launch  --nsockets_per_node 1  --ncores_per_socket 6 --nproc_per_node 1  --master_port=5024'
running benchmark
+ echo 'running benchmark'
+ python -m bind_launch --nsockets_per_node 1 --ncores_per_socket 6 --nproc_per_node 1 --master_port=5024 train.py --save gnmt_wmt16 --dataset-dir /data --preproc-data-dir /preproc_data --target-bleu 24.0 --epochs 8 --math amp_fp16 --max-length-train 75 --print-freq 10 --train-batch-size 256 --test-batch-size 128 --optimizer FusedAdam --lr 2.0e-3 --warmup-steps 200 --remain-steps 6453 --decay-interval 809 --fused-attention --fused-xentropy --no-log-all-ranks
<0-5,6-11> is invalid
libnuma: Warning: cpu argument 6-11 is out of range

usage: numactl [--all | -a] [--interleave= | -i <nodes>] [--preferred= | -p <node>]
               [--physcpubind= | -C <cpus>] [--cpunodebind= | -N <nodes>]
               [--membind= | -m <nodes>] [--localalloc | -l] command args ...
       numactl [--show | -s]
       numactl [--hardware | -H]
       numactl [--length | -l <length>] [--offset | -o <offset>] [--shmmode | -M <shmmode>]
               [--strict | -t]
               [--shmid | -I <id>] --shm | -S <shmkeyfile>
               [--shmid | -I <id>] --file | -f <tmpfsfile>
               [--huge | -u] [--touch | -T]
               memory policy | --dump | -d | --dump-nodes | -D

memory policy is --interleave | -i, --preferred | -p, --membind | -m, --localalloc | -l
<nodes> is a comma delimited list of node numbers or A-B ranges or all.
Instead of a number a node can also be:
  netdev:DEV the node connected to network device DEV
  file:PATH  the node the block device of path is connected to
  ip:HOST    the node of the network device host routes through
  block:PATH the node of block device path
  pci:[seg:]bus:dev[:func] The node of a PCI device
<cpus> is a comma delimited list of cpu numbers or A-B ranges or all
all ranges can be inverted with !
all numbers and ranges can be made cpuset-relative with +
the old --cpubind argument is deprecated.
use --cpunodebind or --physcpubind instead
<length> can have g (GB), m (MB) or k (KB) suffixes
+ ret_code=0
+ set +x
ENDING TIMING RUN AT 2020-06-09 07:18:01 PM
RESULT,RNN_TRANSLATOR,,4,nvidia,2020-06-09 07:17:57 PM

~/training_results_v0.6/NVIDIA/benchmarks/gnmt/implementations/pytorch$ cat config_SUT.sh
#!/bin/bash

## System run parms
DGXNNODES=1
DGXSYSTEM=$(basename $(readlink -f ${BASH_SOURCE[0]}) | sed 's/^config_//' | sed 's/\.sh$//' )
WALLTIME=5:00:00

## DL params
LR=${LR:-"2.0e-3"}
TRAIN_BATCH_SIZE=${TRAIN_BATCH_SIZE:-256}
TEST_BATCH_SIZE=${TEST_BATCH_SIZE:-128}
WARMUP_STEPS=${WARMUP_STEPS:-200}
REMAIN_STEPS=${REMAIN_STEPS:-6453}
DECAY_INTERVAL=${DECAY_INTERVAL:-809}
TARGET=${TARGET:-24.0}
MAX_SEQ_LEN=${MAX_SEQ_LEN:-75}
NUMEPOCHS=${NUMEPOCHS:-8}
MATH=${MATH:-amp_fp16}
EXTRA_OPTS=${EXTRA_OPTS-"\
   --fused-attention \
   --fused-xentropy \
   --no-log-all-ranks \
   "}

## System config params
DGXNGPU=1
DGXSOCKETCORES=6
DGXHT=2 	# HT is on is 2, HT off is 1
DGXIBDEVICES=''
DGXNSOCKET=1
BIND_LAUNCH=1

Question: Is there a missing part on my configuration that's preventing the gpu to be utilized?

**Comment by ipvirg** [comment]
any thoughts from the community?

**Comment by nvpohanh** [comment]
@jonathan-cohen-nvidia Could you help? Thanks

**Comment by nileshnegi** [comment]
From the logs it seems the launch script fails with a numactl CPU out-of-range warning. Can you check the ```lscpu``` output and verify if you have a single socket of 6 cores with hyper-threading enabled?

**Comment by ipvirg** [comment]
lscpu output
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                6
On-line CPU(s) list:   0-5
Thread(s) per core:    1
Core(s) per socket:    6
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 63
Model name:            Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz
Stepping:              2
CPU MHz:               2596.988
BogoMIPS:              5193.97
Hypervisor vendor:     Microsoft
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              30720K
NUMA node0 CPU(s):     0-5
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology cpuid pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm invpcid_single pti fsgsbase bmi1 avx2 smep bmi2 erms invpcid xsaveopt md_clear

**Comment by nileshnegi** [comment]
Hyper-threading is disabled on your system -- Thread(s) per core: 1 and consequently On-line CPU(s) list: 0-5 in your lscpu output. Hence you are getting this error ```libnuma: Warning: cpu argument 6-11 is out of range```

You can retry by replacing with ```DGXHT=1``` in your **System config params** (or) rebooting your system with HT enabled (BIOS setting)

**Comment by ipvirg** [comment]
Made DGXHT=1 change and rerun the training test and still getting the same error.

**Comment by nileshnegi** [comment]
You are using docker, right? In which case, are you mounting the local config files inside your container? If no, re-building the docker image should help - the config file in your current image would not have reflected your change of DGXHT=1

**Comment by ipvirg** [comment]
Yes, I am using docker and I did rebuild the docker image and verify hyper threading disabled  and still getting the same error.

root@26eb22023c99:/workspace/rnn_translator# cat config_SUT.sh | grep HT
DGXHT=1 	# HT is on is 2, HT off is 1
root@26eb22023c99:/workspace/rnn_translator#

**Comment by ipvirg** [comment]
I switched to Azure Standard NC6s_v3 instance and resolved the issue.
